### ISSUE TITLE
Fetch map selector images based on screen resolution

### ISSUE THREAD

**Issue by tsa96** [momentum-mod/game#2112]
Game sibling of https://github.com/momentum-mod/website/issues/907. Map data from the backend (both paginated endpoint and flat file) include map images as arrays of 
```
{
  id: <some uuid>,
  small: <that uuid>-small.jpg,
  medium: <that uuid>-medium.jpg,
  large: <that uuid>-large.jpg,
  xl: <that uuid>-xl.jpg (coming soon!)
}
```
with XL version coming soon. Map cache code should check the user's current resolution and use that to pick the type it downloads.

I'm working on map cache atm, so will probably be able to do this.

**Comment by Gocnak** [comment]
Why not have the game download XL and resize locally? Does the site use the various sizes?

**Comment by tsa96** [comment]
Yeah, site alone warrants doing the resizing (also we're going to doing some image transformations anyway as we convert from PNG to JPEG using `mozjpeg`).

We'd probably want to actually cache the resized image, not the original, since we want to keep the game image cache as low as possible - no point storing a 2560x1440 image for someone with a 1920x1080 monitor. So simpler to just fetch the appropriate sized image than do on the client.

Also worth noting that current (old backend) JPEGs we store are converted to JPEG with a quality of `80`, which is very low and results in severe artifacting. New backend uses `90`, file size for 1920x1080 seems to increase by about 50% on average.

**Comment by tsa96** [comment]
Already did this, last November.
